tests/models/test_momentum_balance.py

```diff
@@ -289,3 +293,396 @@ def test_lithostatic(dim: int):
+            corresponding non-nan expected values.
+        tols: ``len=4``
+
+            Tolerances for the comparisons. The first two are for the elastic and
+            plastic displacement jumps, respectively, the third is for the displacement
+            in the cells above the fracture and the fourth is for the traction on the
+            fracture.
+        compare_means: Whether to compare the means of the computed values to the
+            expected values. If False, the values are compared element-wise.
+
+    Returns:
+        Tuple of the computed elastic displacement jump, plastic displacement jump,
+        displacement in the cells above the fracture and traction on the fracture.
+
+
+    """
+    nd = setup.nd  # Shorthand for number of dimensions.
+    matrix = setup.mdg.subdomains(dim=nd)[0]
+    fractures = setup.mdg.subdomains(dim=nd - 1)
+    assert len(fractures) == 1  # Below code assumes a single fracture.
+    fracture = fractures[0]
+
+    # Get plastic and elastic displacement jumps on the fracture in local coordinates.
+    u_p_loc = setup.plastic_displacement_jump(fractures).value(setup.equation_system)
+    u_e_loc = setup.elastic_displacement_jump(fractures).value(setup.equation_system)
+
+    # Transform the jumps to global coordinates and corresponding to the j side of the
+    # fracture being the one with the lower y-coordinate (jumps are k-j).
+
+    # First, rotate the local displacements to global coordinates. Then, switch sign if
+    # the fracture normal points downwards. This is needed to counteract the cases when
+    # the j ("left") side of the fracture is the one with the higher y-coordinate, i.e.
+    # the upper half of the domain.
+    proj = setup.mdg.subdomain_data(fracture)["tangential_normal_projection"]
+    n = proj.normals
+    rot = proj.project_tangential_normal().T
+    u_p = rot @ u_p_loc
+    u_e = rot @ u_e_loc
+
+    sign = np.tile(np.sign(n[1]), (setup.nd, 1)).ravel("F")
+    u_p = (sign * u_p).reshape((nd, -1), order="F")
+    u_e = (sign * u_e).reshape((nd, -1), order="F")
+
+    u_domain = (
+        setup.displacement([matrix])
+        .value(setup.equation_system)
+        .reshape((nd, -1), order="F")
+    )
+    u_top = u_domain[:, matrix.cell_centers[1] > 0.5]
+    # Compare the computed values to the expected values.
+    if compare_means:
+        assert np.allclose(np.mean(u_e, axis=1), u_e_expected, rtol=tols[0])
+    else:
+        assert np.allclose(u_e, np.reshape(u_e_expected, (nd, 1)), rtol=tols[0])
+
+    if compare_means:
+        assert np.allclose(np.mean(u_p, axis=1), u_p_expected, rtol=tols[1])
+    else:
+        assert np.allclose(u_p, np.reshape(u_p_expected, (nd, 1)), rtol=tols[1])
+
+    # Matrix displacement above the fracture. We only compare those values that are not
+    # nan in the expected values.
+    mask_u = np.logical_not(np.isnan(u_top_expected))
+    if compare_means:
+        expected = np.array(u_top_expected)[mask_u]
```

Comment:
Why redeclear as an array here?

Reply:
Has to do with indexing. Alternative is to force the parameter to be array, but that leads to more writing when calling the function.